test/test_common/utility.cc

```diff
@@ -26,18 +26,21 @@
 #include "test/test_common/printers.h"
 
 #include "absl/strings/string_view.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
 #include "gtest/gtest.h"
 
 using testing::GTEST_FLAG(random_seed);
 
 namespace Envoy {
 
-static const int32_t SEED = std::chrono::duration_cast<std::chrono::nanoseconds>(
-                                std::chrono::system_clock::now().time_since_epoch())
-                                .count();
+int32_t getSeed() {
+  static const int32_t seed = absl::ToUnixNanos(absl::Now());
```

Comment:
I see. The old code was probably only working because it got 'lucky', despite its static-init fiasco.  With absl::time you are getting less lucky. 

Can you try just removing the 'static' keyword here? It will only get called once (per test) when TestRandomGenerator is constructed, and that should be fine.

Reply:
@jmarantz I'm reviewing https://github.com/envoyproxy/envoy/pull/4314 atm and I tend to think it would be better to make this `static`. `TestRandomGenerator` can be constructed more than once per test; it's just a stack or member variable, it seems safer to ensure that it has a consistent seed each time.

Reply:
Actually I followed up in #4314 as that's where the code change is being proposed; hopefully there's enough context there now.